### DISCUSSION TITLE
Request to verify it with TLS 1.3 ciphers and check on Renci.SshNet.dll

### DISCUSSION THREAD

**Discussion by ronakt32** [sshnet/SSH.NET#993] [Q&A]
Helo Team,

RHEL 8.6 is not supporting the SSH tunnel via Renci.SshNet.dll, possibly because of the latest TLS 1.3 ciphers

This happens when old communication is via SSL and TLS 1.1 is disabled

Is it possible to update with the latest ciphers and rebuilt the project?

TLS	v1.3	supports	5 cipher suites.	
– TLS_AES_128_GCM_SHA256
– TLS_AES_256_GCM_SHA384
– TLS_CHACHA20_POLY1305_SHA256
– TLS_AES_128_CCM_SHA256
– TLS_AES_128_CCM_8_SHA256

current dll version we are using is: 

![image](https://user-images.githubusercontent.com/111415438/185052556-803941f8-f3bc-4e63-903c-bb4dd19b6859.png)

**Comment by darkoperator** [comment]
SSH Ciphers are separare from TLS. SSH does not use SSL, different protocols. Under your sshd.conf check HostKeyAlgorithms and KexAlgorithms for the ciphers set for your server

**Comment by ronakt32** [comment]
Thank you for your reply,

Following are the Ciphers in my RHEL 8.6 server under sshd.conf

Ciphers aes128-ctr,aes192-ctr,aes256-ctr,aes128-cbc,3des-cbc,aes192-cbc,aes256-cbc
MACs hmac-sha2-512,hmac-sha2-256,hmac-sha1

I believe they or some of them are not supported by Renci.SshNet.dll, how can we make them support it?
Please help!

**Comment by darkoperator** [comment]
Not the developer but I believe those are supported. What error did you started having? Anything on the logs of the Linux server?

**Comment by ronakt32** [comment]
@darkoperator sorry was a bit away from the desk, the following is an error generated : 

Logs of Working RHEL 8.6 server with SSL TLSv1, TLSv1.2 TLSv1.3

06:20:56:457 - Start SSH Tunneling...
06:20:56:474 - SSH Tunneling -> IsVPNClientStartedForServer...
06:20:56:667 - SSH Tunneling -> IsVPNClientStarted start...
06:20:56:668 - lClientVersion -> 1
06:20:56:674 - SSH Tunneling -> StartVPNClient5...
06:20:56:674 - ForwardingRemoteIPAddress-> 10.10.2.233
06:20:56:676 - SSHServerIPAddress -> 10.10.2.251
06:20:56:677 - SSHServerPort -> 22
06:20:56:678 - SSHServerUserName -> root
06:20:57:140 - Tunnel Successful -> ForwardingLocalIP : 127.0.0.188
06:20:57:142 - Tunnel Successful -> ForwardingRemoteIPAddress : 10.10.2.233
06:20:57:143 - Tunnel Successful -> ForwardingLocalPort : 7897
06:20:57:145 - Tunnel Successful -> ForwardingRemotePort : 3389

Logs of Not Working with RHEL 8.6 server with TLSv1.2 TLSv1.3

08:33:56:037 - Start SSH Tunneling...
08:33:56:038 - SSH Tunneling -> IsVPNClientStartedForServer...
08:33:56:190 - SSH Tunneling -> IsVPNClientStarted start...
08:33:56:191 - lClientVersion -> 1
08:33:56:198 - SSH Tunneling -> StartVPNClient5...
08:33:56:199 - ForwardingRemoteIPAddress-> 10.10.2.233
08:33:56:200 - SSHServerIPAddress -> 10.10.2.251
08:33:56:201 - SSHServerPort -> 22
08:33:56:202 - SSHServerUserName -> root
08:33:56:891 - Error Tunnel StackTrace -    at Renci.SshNet.Session.WaitOnHandle(WaitHandle waitHandle, TimeSpan timeout)
   at Renci.SshNet.Session.Connect()
   at Renci.SshNet.BaseClient.Connect()
08:33:56:893 - Error Tunnel InnerException - 
08:33:56:894 - Error Tunnel GetBaseException - Renci.SshNet.Common.SshConnectionException: An established connection was aborted by the server.
   at Renci.SshNet.Session.WaitOnHandle(WaitHandle waitHandle, TimeSpan timeout)
   at Renci.SshNet.Session.Connect()
   at Renci.SshNet.BaseClient.Connect()
08:37:41:436 - SSH Tunneling -> IsVPNClientStartedForServer...
08:37:41:462 - SSH Tunneling -> IsVPNClientStarted start...
08:37:41:463 - lClientVersion -> 2
08:37:41:470 - SSH Tunneling -> StartVPNClient5...
08:37:41:470 - ForwardingRemoteIPAddress-> 192.168.0.11
08:37:41:471 - SSHServerIPAddress -> 10.10.2.251
08:37:41:472 - SSHServerPort -> 22
08:37:41:474 - SSHServerUserName -> root
08:37:43:302 - Error Tunnel StackTrace -    at Renci.SshNet.Session.WaitOnHandle(WaitHandle waitHandle, TimeSpan timeout)
   at Renci.SshNet.Session.WaitOnHandle(WaitHandle waitHandle)
   at Renci.SshNet.Session.Connect()
   at Renci.SshNet.BaseClient.Connect()
08:37:43:303 - Error Tunnel InnerException - 
08:37:43:304 - Error Tunnel GetBaseException - Renci.SshNet.Common.SshConnectionException: An established connection was aborted by the server.
   at Renci.SshNet.Session.WaitOnHandle(WaitHandle waitHandle, TimeSpan timeout)
   at Renci.SshNet.Session.WaitOnHandle(WaitHandle waitHandle)
   at Renci.SshNet.Session.Connect()
   at Renci.SshNet.BaseClient.Connect()
